smt/sampling_methods/lhs.py

```diff
@@ -323,3 +323,88 @@ def _ese(self, dim, nt):
             return_hist=True,
         )
         return P
+
+    def expand_lhs(self, x, xlimits, points, method="basic"):
+        """
+        Given a Latin Hypercube Sample (LHS) "x", returns an expanded LHS
+        by adding "points" new points.
+
+        Parameters
+        ----------
+        x : array
+            Initial LHS.
+        xlimits : array
+            Limits for each dimension of the initial sample.
+        points : integer
+            Number of points that are to be added to the expanded LHS.
+        method : str, optional
+            Methodoly for the construction of the expanded LHS.
+            The default is "basic".
+
+        Returns
+        -------
+        x_new : array
+            Expanded LHS.
+
+        """
+
+        new_num = len(x) + points
+        if new_num % len(x) != 0:
+            print(
+                "WARNING: The added number of points is not a "
+                "multiple of the initial number of points."
+                "Thus, it cannot be ensured that the output is an LHS."
+            )
+
+        # For future functionalities
+        if method == "basic":
+
+            # Evenly spaced intervals with the final dimension of the LHS
+            intervals = []
+            for i in range(len(xlimits)):
+                intervals.append(np.linspace(xlimits[i][0], xlimits[i][1], new_num + 1))
+
+            # Creates a subspace with the rows and columns that have no points
+            # in the new space
+            subspace_limits = [[]] * len(xlimits)
+            subspace_bool = []
+            for i in range(len(xlimits)):
+                subspace_limits[i] = []
+
+                subspace_bool.append(
+                    [
+                        [
+                            intervals[i][j] < x[kk][i] < intervals[i][j + 1]
+                            for kk in range(len(x))
+                        ]
+                        for j in range(len(intervals[i]) - 1)
+                    ]
+                )
+
+                [
+                    subspace_limits[i].append([intervals[i][ii], intervals[i][ii + 1]])
+                    for ii in range(len(subspace_bool[i]))
+                    if not True in subspace_bool[i][ii]
```

Comment:
In this case I think it should be `not True` instead of `False`. I have a boolean list and what I want is to continue if there are no `True` in the whole list. If I change it to `False`, the code will continue if there is just one `False` in the entire list.

Reply:
I am not sure to understand: a comprehension list will scan all the elements whatever the `if condition`. There is no continue or not question. Here you just append the element if the condition is `True` 
Besides now I see that the precedence of operators is not that obvious is it: 
`if (not True) in subspace_bool[i][ii]` (in this case `not True` == `False`)
or
`if not (True in subspace_bool[i][ii])` 
Please add parentheses to make it explicit. 
I would even say that in this case for maintainance (future you?!) it would be better to write explicit for loops instead of comprehension lists as the expressions are rather long.

Reply:
`if not (True in subspace_bool[i][ii]) `  That conditions works too, I can change it.

At first, I've tried to make explicit loops and the problem was the computational time. I know it's a bit hard to understand like that but one of the goals was to make the code as fast as we could.